internal/node/node.bzl

```diff
@@ -35,6 +35,11 @@ def _trim_package_node_modules(package_name):
         segments += [n]
     return "/".join(segments)
 
+NodeJSSourcesInfo = provider(
+    doc = """Minimal files needed to run a NodeJS program, without npm packages""",
```

Comment:
I started a doc string here but it needs to answer more questions

- how is this intended to be used?
- where do the npm packages come from in the consuming end?

Reply:
Good point about "where do the npm packages come from" - in the new way of having the fine-grained node modules we indeed need to provide these here in the provider also. I updated that.

Not sure what exactly you mean with "    * how is this intended to be used", but I did add a sentence for it.